.goreleaser.yaml

```diff
@@ -0,0 +1,90 @@
+builds:
+  - main: ./cmd/auth-provider-gcp
+    id: auth-provider-gcp
+    no_unique_dist_dir: true
+    binary: auth-provider-gcp/{{.Os}}-{{.Arch}}/{{.Tag}}/auth-provider-gcp
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - darwin
+      - windows
+    goarch:
+      - amd64
+      - arm64
+    ldflags:
+      - -X k8s.io/component-base/version/verflag.programName=auth-provider-gcp
+
+      - -X k8s.io/component-base/version.buildDate={{.Env.BUILD_DATE}}
+      - -X k8s.io/component-base/version.gitCommit={{.Env.KUBE_GIT_COMMIT}}
+      - -X k8s.io/component-base/version.gitMajor={{.Env.KUBE_GIT_MAJOR}}
+      - -X k8s.io/component-base/version.gitMinor={{.Env.KUBE_GIT_MINOR}}
+      - -X k8s.io/component-base/version.gitTreeState={{.Env.KUBE_GIT_TREE_STATE}}
+      - -X k8s.io/component-base/version.gitVersion={{.Env.KUBE_GIT_VERSION}}
+
+      - -X k8s.io/client-go/pkg/version.buildDate={{.Env.BUILD_DATE}}
+      - -X k8s.io/client-go/pkg/version.gitCommit={{.Env.KUBE_GIT_COMMIT}}
+      - -X k8s.io/client-go/pkg/version.gitMajor={{.Env.KUBE_GIT_MAJOR}}
+      - -X k8s.io/client-go/pkg/version.gitMinor={{.Env.KUBE_GIT_MINOR}}
+      - -X k8s.io/client-go/pkg/version.gitTreeState={{.Env.KUBE_GIT_TREE_STATE}}
+      - -X k8s.io/client-go/pkg/version.gitVersion={{.Env.KUBE_GIT_VERSION}}
+
+  - main: ./cmd/gke-exec-auth-plugin
+    id: gke-exec-auth-plugin
+    no_unique_dist_dir: true
+    binary: gke-exec-auth-plugin/{{.Os}}-{{.Arch}}/{{.Tag}}/gke-exec-auth-plugin
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - darwin
+      - windows
+    goarch:
+      - amd64
+      - arm64
+    ldflags:
+      - -X k8s.io/component-base/version/verflag.programName=gke-exec-auth-plugin
+
+      - -X k8s.io/component-base/version.buildDate={{.Env.BUILD_DATE}}
+      - -X k8s.io/component-base/version.gitCommit={{.Env.KUBE_GIT_COMMIT}}
+      - -X k8s.io/component-base/version.gitMajor={{.Env.KUBE_GIT_MAJOR}}
+      - -X k8s.io/component-base/version.gitMinor={{.Env.KUBE_GIT_MINOR}}
+      - -X k8s.io/component-base/version.gitTreeState={{.Env.KUBE_GIT_TREE_STATE}}
+      - -X k8s.io/component-base/version.gitVersion={{.Env.KUBE_GIT_VERSION}}
+
+      - -X k8s.io/client-go/pkg/version.buildDate={{.Env.BUILD_DATE}}
+      - -X k8s.io/client-go/pkg/version.gitCommit={{.Env.KUBE_GIT_COMMIT}}
+      - -X k8s.io/client-go/pkg/version.gitMajor={{.Env.KUBE_GIT_MAJOR}}
+      - -X k8s.io/client-go/pkg/version.gitMinor={{.Env.KUBE_GIT_MINOR}}
+      - -X k8s.io/client-go/pkg/version.gitTreeState={{.Env.KUBE_GIT_TREE_STATE}}
+      - -X k8s.io/client-go/pkg/version.gitVersion={{.Env.KUBE_GIT_VERSION}}
+
+  - main: ./cmd/gke-gcloud-auth-plugin
+    id: gke-gcloud-auth-plugin
+    no_unique_dist_dir: true
+    binary: gke-gcloud-auth-plugin/{{.Os}}-{{.Arch}}/{{.Tag}}/gke-gcloud-auth-plugin
```

Comment:
These paths are surprising, but I'm trying to mirror the current GCS structure.

I would have thought that this structure would be more logical:


`{{.Tag}}/{{.Os}}-{{.Arch}}/gke-gcloud-auth-plugin`

or 

`gke-gcloud-auth-plugin/{{.Tag}}/{{.Os}}-{{.Arch}}/gke-gcloud-auth-plugin`

But in general the current structure implies to me that we're tagging the different architectures separately, which is ... not what we're doing.

Reply:
@ndixita @BenTheElder do you remember why we ended with these structures?

Reply:
I'm going to switch to something that is more conventional: project name (because we want to think of these as separate projects, even though they share a repo), then tag, then architecture.

The current staging buckets are basically broken anyway, they upload the binaries named by SHA, which I don't think work with our promotion tools or any tools I've seen in the ecosystem.